eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -48,6 +48,52 @@ class MultiPartHandler(nodeParams: NodeParams, db: IncomingPaymentsDb, commandBu
    */
   def doHandle(paymentHash: ByteVector32): Boolean = true
 
+  /**
+   * Fulfill all parts of an incoming payment.
+   * Can be overridden for a more fine-grained control of when and how to fulfill.
+   */
+  def doFulfill(system: ActorSystem, preimage: ByteVector32, e: MultiPartPaymentFSM.MultiPartPaymentSucceeded)(implicit log: LoggingAdapter): Unit = {
```

Comment:
I'm not comfortable with letting plugin writers override that kind of tricky method as a recommended practice.

How about we let those methods as-is, but don't recommend overriding them, and go akka-style by adding a  `preFulfill` and `postFulfill` methods (the latter replacing `onSuccess`)?

Reply:
I'll try that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1324/commits/7f4545660b41d286a12c2a383f8b9fb23947801d